components/NavigationBar/src/MDCNavigationBar.m

```diff
@@ -460,10 +460,13 @@ - (CGRect)mdc_frameAlignedVertically:(CGRect)frame
 - (CGRect)mdc_frameAlignedHorizontally:(CGRect)frame
                              alignment:(MDCNavigationBarTitleAlignment)alignment {
   switch (alignment) {
-    case MDCNavigationBarTitleAlignmentCenter:
-      return CGRectMake(CGRectGetMaxX(self.bounds) / 2 - frame.size.width / 2, frame.origin.y,
-                        frame.size.width, frame.size.height);
-
+    case MDCNavigationBarTitleAlignmentCenter: {
+      CGFloat xOrigin = CGRectGetMaxX(self.bounds) / 2 - frame.size.width / 2;
+      if (frame.origin.x <= xOrigin) {
+        return CGRectMake(xOrigin, frame.origin.y,
+                          frame.size.width, frame.size.height);
+      }
+    }
```

Comment:
Are you deliberately falling through to the next case here? Intentional falling through [needs a comment](https://engdoc.corp.google.com/eng/doc/objcguide.xml?external=y&showone=Conditionals#Conditionals).